### PR TITLE
Add health status parameter to cat indices API

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds a health status parameter to the cat indices API for
filtering on indices that match the specified status (green|yellow|red).

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.indices/10_basic.yaml [hunk 1]
```diff
@@ -71,6 +71,41 @@
...
+          settings:
+            number_of_shards: "1"
+            number_of_replicas: "0"
+  - do:
+      indices.create:
+        index: bar
+        body:
+          settings:
+            number_of_shards: "1"
+            number_of_replicas: "1"
```
Comment (javanna) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.indices/10_basic.yaml]: I think this assumes that no rest test runner runs against multiple nodes. I think the some of the client test runners do though, so this index will be green for them.

**Comment by javanna** [comment]
I left a comment around testing, LGTM otherwise

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.indices/10_basic.yaml (hunk 1 above)
Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.indices/10_basic.yaml]: That is a very good catch, thank you @javanna.

Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.indices/10_basic.yaml]: @javanna I pushed b0ad122d52aadbe2bd9530dfa0b92d8bd7043fe8, can you take a look?

**Comment by javanna** [comment]
LGTM thanks @jasontedor

**Comment by jasontedor** [comment]
Thanks @javanna.

**Pull request merged**